client/state/themes/actions/confirm-delete.js

```diff
@@ -24,7 +25,22 @@ import 'calypso/state/themes/init';
  */
 export function confirmDelete( themeId, siteId ) {
 	return ( dispatch, getState ) => {
-		const { name: themeName } = getTheme( getState(), siteId, themeId );
+		let theme = getTheme( getState(), siteId, themeId );
+		if ( ! theme ) {
+			const recommendedThemes = getRecommendedThemes( getState() );
+			theme = recommendedThemes.find( ( recTheme ) => recTheme.id === themeId );
+		}
+
+		if ( ! theme ) {
```

Comment:
This fallback behavior was missing before, but it could be useful in case there's another scenario we didn't think about, e.g. some different type of theme list we didn't consider.